addons/mail/__manifest__.py

```diff
@@ -51,222 +51,25 @@
     'application': True,
     'assets': {
         'web._assets_primary_variables': [
-            'mail/static/src/scss/variables.scss',
+            'mail/static/src/scss/variables/primary_variables.scss',
         ],
         'web.assets_backend': [
-            'mail/static/src/js/core/translation.js',
-            'mail/static/src/js/many2many_tags_email.js',
-            'mail/static/src/js/m2x_avatar_user.js',
-            'mail/static/src/js/field_char.js',
-            'mail/static/src/js/document_viewer.js',
-            'mail/static/src/js/basic_view.js',
-            'mail/static/src/js/systray/systray_activity_menu.js',
-            'mail/static/src/js/tours/mail.js',
-            'mail/static/src/js/tools/debug_manager.js',
-            'mail/static/src/js/custom_filter_item.js',
-            'mail/static/src/js/utils.js',
-            'mail/static/src/js/activity.js',
-            'mail/static/src/js/views/activity/activity_view.js',
-            'mail/static/src/js/views/activity/activity_model.js',
-            'mail/static/src/js/views/activity/activity_controller.js',
-            'mail/static/src/js/views/activity/activity_renderer.js',
-            'mail/static/src/js/views/activity/activity_record.js',
-            'mail/static/src/js/views/activity/activity_cell.js',
-            'mail/static/src/js/emojis.js',
-            'mail/static/src/js/emojis_mixin.js',
-            'mail/static/src/js/field_emojis_common.js',
-            'mail/static/src/js/field_char_emojis.js',
-            'mail/static/src/js/field_text_emojis.js',
-            'mail/static/src/scss/emojis.scss',
+            'mail/static/src/js/**/*.js',
+            ('remove', 'mail/static/src/js/main.js'), # main is prod only
```

Comment:
I would move `main.js` at the root of `src` so that we don't need this remove. We shouldn't have "type" folders in the first place anyway, this `/js` should be considered legacy.